salt/modules/kubernetes.py

```diff
@@ -64,16 +80,31 @@ def _setup_conn(**kwargs):
                                      'http://localhost:8080')
     username = __salt__['config.option']('kubernetes.user')
     password = __salt__['config.option']('kubernetes.password')
+    ca_cert = __salt__['config.option']('kubernetes.certificate-authority-data')
+    client_cert = __salt__['config.option']('kubernetes.client-certificate-data')
+    client_key = __salt__['config.option']('kubernetes.client-key-data')
+    ca_cert_file = __salt__['config.option']('kubernetes.certificate-authority-file')
+    client_cert_file = __salt__['config.option']('kubernetes.client-certificate-file')
+    client_key_file = __salt__['config.option']('kubernetes.client-key-file')
 
     # Override default API settings when settings are provided
-    if kwargs.get('api_url'):
-        host = kwargs['api_url']
+    if 'api_url' in kwargs:
```

Comment:
We don't need all of these conditionals. We can just use the `.get()` each time and it will default to returning `None` if the supplied value is not found.